engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java

```diff
@@ -54,12 +53,12 @@
 @Persistable(clazz = GenericAnnotations$.class, id = 1216)
 @Persistable(clazz = ExpressionAnnotations$.class, id = 1217)
 @Persistable(clazz = FullyQualifiedNames$.class, id = 1218)
-@Persistable(clazz = AliasAnalysis$Info$Occurrence.class, id = 1261, allowInlining = false)
-@Persistable(clazz = AliasAnalysis$Info$Scope$Root.class, id = 1262, allowInlining = false)
-@Persistable(clazz = AliasAnalysis$Info$Scope$Child.class, id = 1263, allowInlining = false)
-@Persistable(clazz = AliasAnalysis$Graph$Occurrence$Use.class, id = 1264, allowInlining = false)
-@Persistable(clazz = AliasAnalysis$Graph$Occurrence$Def.class, id = 1265, allowInlining = false)
-@Persistable(clazz = AliasAnalysis$Graph$Link.class, id = 1266, allowInlining = false)
+@Persistable(clazz = AliasAnalysisInfo.Occurrence.class, id = 1261, allowInlining = false)
+@Persistable(clazz = AliasAnalysisInfo$Scope$Root.class, id = 1262, allowInlining = false)
```

Comment:
Shouldn't these four be also unnested?

Reply:
I've tried
```diff
Index: engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java b/engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java
--- a/engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java	(revision 40381e8ff7262b88bd8da8a3744155386c198be6)
+++ b/engine/runtime-compiler/src/main/java/org/enso/compiler/pass/analyse/PassPersistance.java	(date 1710521833168)
@@ -54,10 +54,10 @@
 @Persistable(clazz = ExpressionAnnotations$.class, id = 1217)
 @Persistable(clazz = FullyQualifiedNames$.class, id = 1218)
 @Persistable(clazz = AliasAnalysisInfo.Occurrence.class, id = 1261, allowInlining = false)
-@Persistable(clazz = AliasAnalysisInfo$Scope$Root.class, id = 1262, allowInlining = false)
-@Persistable(clazz = AliasAnalysisInfo$Scope$Child.class, id = 1263, allowInlining = false)
-@Persistable(clazz = AliasAnalysisGraph$Occurrence$Use.class, id = 1264, allowInlining = false)
-@Persistable(clazz = AliasAnalysisGraph$Occurrence$Def.class, id = 1265, allowInlining = false)
+@Persistable(clazz = AliasAnalysisInfo.Scope$Root.class, id = 1262, allowInlining = false)
+@Persistable(clazz = AliasAnalysisInfo.Scope$Child.class, id = 1263, allowInlining = false)
+@Persistable(clazz = AliasAnalysisGraph.Occurrence$Use.class, id = 1264, allowInlining = false)
+@Persistable(clazz = AliasAnalysisGraph.Occurrence$Def.class, id = 1265, allowInlining = false)
 @Persistable(clazz = AliasAnalysisGraph.Link.class, id = 1266, allowInlining = false)
 public final class PassPersistance {
   private PassPersistance() {}
```
but I'm getting
```
[error] X:\NBO\enso\engine\runtime-compiler\src\main\java\org\enso\compiler\pass\analyse\PassPersistance.java:62:1:  error: There should be exactly one constructor in org.enso.compiler.pass.analyse.PassPersistance
[error] public final class PassPersistance {
[error]              ^
[error] X:\NBO\enso\engine\runtime-compiler\src\main\java\org\enso\compiler\pass\analyse\PassPersistance.java:57:1:  error: cannot find symbol
[error] @Persistable(clazz = AliasAnalysisInfo.Scope$Root.class, id = 1262, allowInlining = false)
[error]                                       ^  symbol:   class Scope$Root
[error]   location: interface AliasAnalysisInfo
[error] X:\NBO\enso\engine\runtime-compiler\src\main\java\org\enso\compiler\pass\analyse\PassPersistance.java:59:1:  error: cannot find symbol
[error] @Persistable(clazz = AliasAnalysisGraph.Occurrence$Use.class, id = 1264, allowInlining = false)
[error]                                        ^  symbol:   class Occurrence$Use
```

It feels like if there are any `$` in the path, then the whole path needs to be `$`. Am I wrong?

Reply:
> It feels like if there are any $ in the path, then the whole path needs to be $. Am I wrong?

I didn't get this comment. Basically you can't have more than 2-level nesting if you don't want `$` in the name. `Use` or `Def` are at the 3rd level.

Reply:
@radeusgd Note that IntelliJ will not give you proper symbol completion / navigation capabilities when trying to call some Scala code from Java. When dealing with this, the most reliable way to see what is going on is to directly look into the compiled `class` files, for example:
```
> javap engine/runtime-compiler/target/scala-2.13/classes/org/enso/compiler/pass/analyse/AliasAnalysisGraph$Occurrence$Def.class
Compiled from "AliasAnalysisGraph.scala"
public class org.enso.compiler.pass.analyse.AliasAnalysisGraph$Occurrence$Def implements org.enso.compiler.pass.analyse.AliasAnalysisGraph$Occurrence,scala.Product {
  public scala.collection.Iterator<java.lang.String> productElementNames();
  public int id();
  public java.lang.String symbol();
  public java.util.UUID identifier();
  public scala.Option<java.util.UUID> externalId();
  public boolean isLazy();
  public org.enso.compiler.pass.analyse.AliasAnalysisGraph$Occurrence$Def copy(int, java.lang.String, java.util.UUID, scala.Option<java.util.UUID>, boolean);
  public int copy$default$1();
  public java.lang.String copy$default$2();
  public java.util.UUID copy$default$3();
  public scala.Option<java.util.UUID> copy$default$4();
  public boolean copy$default$5();
  public java.lang.String productPrefix();
  public int productArity();
  public java.lang.Object productElement(int);
  public scala.collection.Iterator<java.lang.Object> productIterator();
  public boolean canEqual(java.lang.Object);
  public java.lang.String productElementName(int);
  public int hashCode();
  public java.lang.String toString();
  public boolean equals(java.lang.Object);
  public org.enso.compiler.pass.analyse.AliasAnalysisGraph$Occurrence$Def(int, java.lang.String, java.util.UUID, scala.Option<java.util.UUID>, boolean);
}
```